src/Web/Tailwind.hs

```diff
@@ -101,26 +113,21 @@ instance Default Css where
       |]
 
 instance Text.Show.Show TailwindConfig where
-  show (decodeUtf8 . encode -> config) =
-    -- Use `Object.assign` to merge JSON (produced in Haskell) with the rest of
-    -- config (defined by raw JS; that cannot be JSON encoded)
-    toString
-      [text|
-      module.exports =
-        Object.assign(
-          ${config},
-          {
-            theme: {
-              extend: {},
-            },
-            plugins: [
-              require('@tailwindcss/typography'),
-              require('@tailwindcss/forms'),
-              require('@tailwindcss/line-clamp'),
-              require('@tailwindcss/aspect-ratio')
-            ],
-          })
-      |]
+  show TailwindConfig{..} =
+    toString [text|
+        module.exports = {
+            content: ${content},${strPlugins}
+        }
+        |]
+    where
+      content = Text.pack $ show _tailwindConfigContent
+      strPlugins = if null _tailwindConfigPlugins
+        then ""
+        else "\nplugins: [\n    "
+          <> Text.intercalate ",\n    " (mkJSStrPlugin <$> _tailwindConfigPlugins)
+          <> "\n]"
+      mkJSStrPlugin plugin =
+          "require('@tailwindcss/" <> Text.pack (show plugin) <> "')"
```

Comment:
Even if this is correct, I feel uncomfortable maintaining it (especially with no automated tests) :-) I'd rather use some sort of templating language. Like [this](https://hackage.haskell.org/package/simple-templates-1.0.0/docs/Web-Simple-Templates-Language.html) (I haven't done any thorough research on it).

Reply:
the json instance was confusing to me, too. It's not json, this is why this looks a bit messy. In my fork I added the "theme.js", too, which makes everything even messier. I thought about representing "theme.js" as an `Aeson.Object`, but no use: it contains JavaScript and not strictly Json.

Replacing the string concatenation with template logic makes everything more readable, but there is not a lot to gain. The actual test needed here would check for correct tailwind/Javascript syntax. In practice, this doesn't matter too much: On my fork, I edit `theme.js` and also a custom `input.css` and any mistake there breaks my styling entirely. Mistakes inside the haskell code are easily spotted (and thus all fixed already).

Reply:
A trivial test would be to check against `show (def :: TailwindConfig)`.

Reply:
and we cann have something a bit more readable like this:

```haskelll
instance Text.Show.Show TailwindConfig where
  show TailwindConfig{..} = toString
      [text|
      module.exports = {
          content : ${strContent},
          theme: ${_taillwindConfigTheme},
          plugins: ${strPlugins}
       }
      |]
  where
    strContent = Text.pack $ show _tailwindConfigContent 
    mkStrPlugin plugin = "require('@tailwindcss/" <> show plugin <> "')"
    strPlugins = Text.intercalate ",\n" $ mkStrPlugin <$> _tailwindConfigPlugins
```

This results in a haskell code that is more readable and a `TailwindConfig` that always contains the keys "theme" and "plugins", regardless of whether or not they have values.

I am not opposed of putting a template there, though. Just trying to understand what the priorities are here.

Reply:
Let's go with `hspec` (with or without `hspec-discover`).

cf. https://github.com/srid/emanote/tree/master/emanote/test